eth/handler.go

```diff
@@ -452,6 +468,18 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 		// Filter out any explicitly requested headers, deliver the rest to the downloader
 		filter := len(headers) == 1
 		if filter {
+			// Check for any responses not matching our whitelist
+			if whitelist := pm.whitelist; whitelist != nil {
+				if expected, ok := (*whitelist)[headers[0].Number.Uint64()]; ok {
+					actual := headers[0].Hash()
+					if !bytes.Equal(expected.Bytes(), actual.Bytes()) {
+						p.Log().Debug("dropping peer with non-matching whitelist block", "number", headers[0].Number.Uint64(), "expected", expected, "actual", actual)
+						return errors.New("whitelist block mismatch")
+					}
+
+					p.Log().Debug("whitelist match", "number", headers[0].Number.Uint64(), "expected", expected)
```

Comment:
Please drop the empty line, no use for it. Please make the log message meaningful.

"Whitelist block verified", Please use `hash`, not "expected". It's a convention throughout the code.

Reply:
Done.